x-pack/auditbeat/module/system/socket/guess/creds.go

```diff
@@ -51,7 +51,7 @@ const (
 )
 
 func init() {
-	if err := Registry.AddGuess(&guessStructCreds{}); err != nil {
+	if err := Registry.AddGuess(func() Guesser { return &guessStructCreds{} }); err != nil {
```

Comment:
All the changes to guesses are required because, turns out, every guess instance could only run once, as some of them keep internal state (counters, etc.). Replacing the registration with a factory method ensures that when config is reloaded a fresh set of guesses is run.

Reply:
I will update https://github.com/elastic/beats/pull/21011 at some point to reflect this and probably get rid of the global registry for Guesses.